### PR TITLE
Switching to Sixpack class and enabling conversion.

### PR DESCRIPTION
**Pull request by weerd**
### What does this PR do?

This PR creates a new `Sixpack` class to help handle all experiments on a page. I worked with @DFurnes to initially stub out the idea. He had suggested trying to move away from storing the experiments in the Redux store and instead having a dedicated singleton class to handle experiments on a page and from this initial pass it seems to work rather well.

Being a singleton, the`Sixpack` class can only be instantiated once and will house all the experiments on a page (that means it keeps track of experiments on a page load, OR switching to another page (like a tabbed nav page, but still need to test this out a bit).

Experiments can be setup via Contentful and will activate an alternative when the page is loaded. In Contentful, an editor can select what "actions" or events to convert the experiment on; if none are specified, then it will convert on any of the possible actions.

### Any background context you want to provide?

This PR only adds the changes to include the `Sixpack` class, some constants, and also adds some new console logging for when a Sixpack test is participated or converted:

sixpack console log

The capacity to execute the conversions will happen in middleware (upcoming PR) which will listen for actions related to signups, posts, etc.

### What are the relevant tickets/cards?

Refs [Pivotal ID #159468419](https://www.pivotaltracker.com/story/show/159468419)

### CODE REVIEW

File: resources/assets/services/Sixpack.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import {
+  SIXPACK_EXPERIMENT_SIGNUP_ACTION,
+  SIXPACK_EXPERIMENT_REPORTBACK_POST_ACTION,
+  SIXPACK_EXPERIMENT_CLICKED_BUTTON_ACTION,
+} from '../constants';
+import { sixpackLog } from '../helpers/loggers';
+
+class Sixpack {
+  constructor() {
+    this.reset();
```
Comment (mendelB) [inline resources/assets/services/Sixpack.js]: Why do we need to reset the `this.experiments` in the constructor, wouldn't it always be empty at this point?

File: resources/assets/constants/index.js [hunk 2]
```diff
@@ -20,3 +20,7 @@ export const NetworkStatus = {
   READY: 7,
   ERROR: 8,
 };
+
+export const SIXPACK_EXPERIMENT_SIGNUP_ACTION = 'signup';
+export const SIXPACK_EXPERIMENT_REPORTBACK_POST_ACTION = 'reportbackPost';
+export const SIXPACK_EXPERIMENT_CLICKED_BUTTON_ACTION = 'clickedButton';
```
[the remark this answers is not in the document]
Comment (DFurnes) [reply resources/assets/constants/index.js]: That makes sense to me. I do sometimes wonder about the developer ergonomics of exporting and importing constants (as opposed to throwing a runtime exception on invalid input or ✨ type checking ✨). But I definitely see the argument in favor of 'em!

File: resources/assets/services/Sixpack.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+  /**
+   * Participate in specified experiment.
+   *
+   * @param  {String} experimentName
+   * @param  {Array}  testAlternatives
+   * @param  {Object} options
+   * @return {Promise}
+   */
+  participate(experimentName, testAlternatives = [], options = {}) {
```
Comment (mendelB) [inline resources/assets/services/Sixpack.js]: Are we removing support for the `trafficFraction` param? (I know you had it as a @TODO on the `participateBeta` you added)

File: resources/assets/services/Sixpack.js (hunk 1 above)
Comment (weerd) [reply resources/assets/services/Sixpack.js]: Yeah, but just in case™ 😉

Comment (weerd) [reply resources/assets/services/Sixpack.js]: Didn't feel strongly about removing it, but if you do feel free to let me know!

File: resources/assets/services/Sixpack.js (hunk 3 above)
Comment (weerd) [reply resources/assets/services/Sixpack.js]: `trafficFraction` can be passed as part of the `options` now. along with `kpi` and any other options 💃

**Comment by weerd** [comment]
> So this Singleton is going to replace the /helpers/experiments, and the redux / experiments middleware?

Yep yep! I'll be cleaning the redux store out and any old code related to the old approach once this new approach is all set!

File: resources/assets/services/Sixpack.js (hunk 1 above)
Comment (mendelB) [reply resources/assets/services/Sixpack.js]: Nah I was just curious if there was some fluke you'd bumped into. I always love me some

File: resources/assets/services/Sixpack.js (hunk 3 above)
Comment (mendelB) [reply resources/assets/services/Sixpack.js]: Oh, nice! Where does that get passed over to `sixpack` though?

File: resources/assets/constants/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (weerd) [reply resources/assets/constants/index.js]: Yeah, if we feel it doesn't end up helping out, we can remove. Just something I wanted to try out for now :)

File: resources/assets/services/Sixpack.js (hunk 3 above)
Comment (weerd) [reply resources/assets/services/Sixpack.js]: That's in the next part coming up. It's going to be a field on the `sixpackExperiment` content type in Contentful 💃

**Pull request merged**